capabilityquerier/pkg/capabilityquerier/capabilityquerier.go

```diff
@@ -158,25 +152,44 @@ func mimeTypesMatch(reqMimeType string, respMimeType string) bool {
 	return false
 }
 
-// responds with the http.Response, whether or not the status code was a 406 (indicating we should try with a different
-// mimetype), and any errors.
-func requestWithMimeType(req *http.Request, mimeType string, client *http.Client) (*http.Response, bool, error) {
+// responds with
+//   http status code
+//   tls version
+//   mime type match
+//   capability statement
+//   error
+func requestWithMimeType(req *http.Request, mimeType string, client *http.Client) (int, string, bool, []byte, error) {
+	var httpResponseCode int
+	var tlsVersion string
+	var capStat []byte
+
+	mimeMatches := false
+
 	req.Header.Set("Accept", mimeType)
+	req.Header.Set("no-cache", "") // ensures that we don't cache the previously retrieved mime type in the response header
 
 	resp, err := client.Do(req)
 	if err != nil {
-		return nil, false, errors.Wrapf(err, "making the GET request to %s failed", req.URL.String())
+		return -1, "", false, nil, errors.Wrapf(err, "making the GET request to %s failed", req.URL.String())
 	}
 
-	if resp.StatusCode == http.StatusNotAcceptable {
-		return nil, true, nil
-	}
+	httpResponseCode = resp.StatusCode
+	if httpResponseCode == http.StatusOK {
+		defer resp.Body.Close()
+		respMimeType := resp.Header.Get("Content-Type")
+		if isJSONMIMEType(respMimeType) {
+			mimeMatches = true
```

Comment:
I think it would be helpful to have a comment here to note that the endpoints return xml by default which is why it's checking for whether its a JSON mime type